lib/backend/backend.go

```diff
@@ -231,6 +234,16 @@ var (
 	noEnd = []byte{0}
 )
 
+// RangeEnd returns end of the range for given key.
+func RangeEnd(key []byte) []byte {
+	return nextKey(key)
+}
+
+// NextResourceKey returns the next possible resource key as a string.
+func NextResourceKey(r types.Resource) string {
+	return string(nextKey([]byte(r.GetName())))
+}
```

Comment:
Nit: I'm a bit hesitant about calling this `NextResourceKey`.  Feel like someone looking at a function named `backend.NextResourceKey` might think this produces the next backend key for the resource.  Maybe call it `NextPaginationKey` or something along those lines?